Parse/src/main/java/com/parse/ParseObject.java

```diff
@@ -361,23 +360,15 @@ public ParseObject(String theClassName) {
           "You must specify a Parse class name when creating a new ParseObject.");
     }
     if (AUTO_CLASS_NAME.equals(theClassName)) {
-      theClassName = getClassName(this.getClass());
+      theClassName = getSubclassingController().getClassName(getClass());
     }
 
     // If this is supposed to be created by a factory but wasn't, throw an exception.
-    if (this.getClass().equals(ParseObject.class) && objectTypes.containsKey(theClassName)
-        && !objectTypes.get(theClassName).isInstance(this)) {
+    if (!getSubclassingController().isTypeValid(theClassName, getClass())) {
       throw new IllegalArgumentException(
           "You must create this type of ParseObject using ParseObject.create() or the proper subclass.");
     }
 
-    // If this is an unregistered subclass, throw an exception.
-    if (!this.getClass().equals(ParseObject.class)
-        && !this.getClass().equals(objectTypes.get(theClassName))) {
-      throw new IllegalArgumentException(
-          "You must register this ParseObject subclass before instantiating it.");
-    }
-
```

Comment:
That specific message? Yes. That error case is still handled by `isTypeValid()`.